### PR TITLE
Pull request for Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by zuhebahmed88091**
#### In this PR:

- Added a column `species` of type `string` to your `animals` table. Modify your `schema.sql` file.
### Insert the following data:
- Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
- Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
- Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
- Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
- Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
- Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
- Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
Added these insertions to your `data.sql` file.
 - - - -
 

- [ ] Inside a transaction updated the `animals` table by setting the `species` column to `unspecified`. Verified that change are made. Then rolled back the change and verified that the `species` columns went back to the state before the transaction.

ss2

### Inside a transaction:
 
- [ ] Updated the animals table by setting the species column to digimon for all animals that have a name ending in mon.
- [ ] Updated the animals table by setting the species column to pokemon for all animals that don't have species already set.
- [ ] Verified that changes were made.
- [ ] Committed the transaction.
- [ ] Verified that changes persist after commit.

ss3

### Inside a transaction:

- [ ]  deleted all records in the animals table, then rolled back the transaction.
- [ ] After the rollback verified if all records in the animals table still existed.

ss4

### Inside a transaction:

- [ ] Deleted all animals born after Jan 1st, 2022.
- [ ] Created a savepoint for the transaction.
- [ ] Updated all animals' weight to be their weight multiplied by -1.
- [ ] Rollbacked to the savepoint.
- [ ] Updated all animals' weights that are negative to be their weight multiplied by -1.
- [ ] Committed the transaction.

ss5-1

ss5-2

- [ ] Added all these transactions to your queries.sql file.

### Queries to answer the following questions:

- [ ] How many animals are there?

ss6

- [ ] How many animals have never tried to escape?

ss7

- [ ] What is the average weight of animals?

ss8

- [ ] Who escapes the most, neutered or not neutered animals?

ss9

- [ ] What is the minimum and maximum weight of each type of animal?

ss10

- [ ] What is the average number of escape attempts per animal type of those born between 1990 and 2000?

ss11

#### FIXED:

- [ ] The weight for the first four animals should be negative, and run queries again to get the right results.

req

req3

#### FIXED:

- [ ] The issues with the data (correcting the negative weights) and rerun queries to get the right results for some of the queries

req4-1

req4-2

Thank you for reviewing my PR!! ☺️

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -1,3 +1,12 @@
 /* Populate database with sample data. */
 
-INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg ) VALUES ('Agumon', '2020-02-03', '0', 'true', '10.23'),('Gabumon', '2018-11-15', '2', 'true', '8'), ('Pikachu', '2021-01-07', '1', 'false', '15.04'), ('Devimon', '2017-05-12', '5', 'true', '11');
\ No newline at end of file
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg ) VALUES ('Agumon', '2020-02-03', '0', 'true', '10.23'),('Gabumon', '2018-11-15', '2', 'true', '8'), ('Pikachu', '2021-01-07', '1', 'false', '15.04'), ('Devimon', '2017-05-12', '5', 'true', '11');
+
+-- DAY 2
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Charmander', '2020-02-08', '0', 'false', '11');
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Plantmon', '2021-11-15', '2', 'true', '5.7');
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Squirtle', '1993-04-02', '3', 'false', '12.13');
```
Comment (mckent05) [inline data.sql]: - Good job done so far, but according to the project requirements, the weight for the first four animals should be negative, please fix this and run all queries again so you get the right results. Please fix this 🙏

File: queries.sql [hunk 2]
```diff
@@ -7,4 +7,43 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
...
+SELECT * FROM animals;
+COMMIT;
+SELECT * FROM animals;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT ROUND(AVG(weight_kg), 2) FROM animals;
+SELECT CASE WHEN neutered = 't' THEN 'Neutered' ELSE 'Not Neutered' END AS neutered_status, SUM(escape_attempts) AS total_escape_attempts FROM animals GROUP BY neutered_status;
+SELECT MIN(weight_kg),MAX(weight_kg),species FROM animals GROUP BY species;
+SELECT ROUND(AVG(escape_attempts), 2) AS avg_escape_attempts, species FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY species;
```
Comment (mckent05) [inline queries.sql:49]: - Good job done so far, please fix the issues with the data (correcting the negative weights) and rerun your queries so you can get the right results for some of the queries. Please fix this 👍 👍

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @zuhebahmed88091, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job updating your animal table 👍 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (zuhebahmed88091) [reply data.sql]: req

### FIXED: 
 The weight for the first four animals should be negative.

File: queries.sql (hunk 2 above)
Comment (zuhebahmed88091) [reply queries.sql:49]: Done

**Review by BregornOriginal — APPROVED** [review]
Hi @zuhebahmed88091 

Your project is complete!  There is nothing else to say other than... it's time to merge it 

Congratulations! 

giphy

## Highlights

✅ Linters are passing
✅ Good commit history
✅ Excellent work implementing queries! 💯 

## Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.
Cheers and Happy coding!:aplauso::aplauso::aplauso:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, remember to tag me in your question so I can receive the notification.
------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)[.](https://airtable.com/shrQAqnBwek5a0O0s)._)

**Pull request merged**